File: src/java/HammingDistance.java
/**
 * 461. 汉明距离
 *
 * @author LJY
 */
public class HammingDistance {

    public int hammingDistance(int x, int y) {
        if(x == y){
            return 0;
        }
        int count = 0;
        int mask = 1;
        for (int i = 0; i < 32; i++) {
            if ((mask & x) != (mask & y)) {
                count++;
            }
            mask <<= 1;
        }
        return count;
    }

}

File: src/scala/HammingDistance.scala
object org.jy.HammingDistance {

  def hammingDistance(x: Int, y: Int): Int = {
    if (x == y) {
      return 0
    }
    var count = 0
    var mask = 1
    for (i <- 0 until 32) {
      if ((x & mask) != (y & mask)) {
        count += 1
      }
      mask <<= 1
    }
    count
  }

}
